public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+} from '@elastic/eui';
+import './styles.scss';
+import {
+  createAugmentVisSavedObject,
+  ISavedAugmentVis,
+  VisLayerExpressionFn,
+} from '../../../../../../src/plugins/vis_augmenter/public';
+import { useDispatch } from 'react-redux';
+import { snakeCase, find, isEmpty, get } from 'lodash';
+import {
+  Field,
+  FieldArray,
+  FieldArrayRenderProps,
+  FieldProps,
+  Form,
+  Formik,
+  FormikHelpers,
+} from 'formik';
+import {
+  createDetector,
+  startDetector,
+} from '../../../../public/redux/reducers/ad';
+import { EmbeddablePanel } from '../../../../../../src/plugins/embeddable/public';
+import './styles.scss';
+import EnhancedAccordion from '../EnhancedAccordion';
+import MinimalAccordion from '../MinimalAccordion';
+import {
+  FEATURE_TYPE,
+  UNITS,
+  Detector,
+} from '../../../../public/models/interfaces';
+import { DataFilterList } from '../../../../public/pages/DefineDetector/components/DataFilterList/DataFilterList';
+import {
+  convertTimestampToNumber,
+  getError,
+  isInvalid,
+  validateDetectorName,
+  validateFeatureName,
+  validatePositiveInteger,
+} from '../../../../public/utils/utils';
+import { CUSTOM_AD_RESULT_INDEX_PREFIX } from '../../../../server/utils/constants';
+import {
+  focusOnFirstWrongFeature,
+  initialFeatureValue,
+} from '../../../../public/pages/ConfigureModel/utils/helpers';
+import { FeaturesFormikValues } from '../../../../public/pages/ConfigureModel/models/interfaces';
+import {
+  getIndices,
+  getMappings,
+} from '../../../../public/redux/reducers/opensearch';
+import {
+  featuresToUIMetadata,
+  formikToFeatureAttributes,
+  formikToFilterQuery,
+} from '../../../../public/pages/ReviewAndCreate/utils/helpers';
+import { FormattedFormRow } from '../../../../public/components/FormattedFormRow/FormattedFormRow';
+import { FeatureAccordion } from '../../../../public/pages/ConfigureModel/components/FeatureAccordion';
+import {
+  AD_DOCS_LINK,
+  MAX_FEATURE_NUM,
+} from '../../../../public/utils/constants';
+
+function AddAnomalyDetector({
+  embeddable,
+  closeFlyout,
+  notification,
+  services,
+  mode,
+  setMode,
+  index,
+  selectedDetectorId,
+  setSelectedDetectorId,
+}) {
+  const dispatch = useDispatch();
+  const [queryText, setQueryText] = useState('');
+  useEffect(() => {
+    const getInitialIndices = async () => {
+      await dispatch(getIndices(queryText));
+    };
+    getInitialIndices();
+    dispatch(getMappings(embeddable.vis.data.aggs.indexPattern.title));
+  }, []);
+
+  const [isShowVis, setIsShowVis] = useState(false);
+  const [accordionsOpen, setAccordionsOpen] = useState({ modelFeatures: true });
+  const [detectorNameFromVis, setDetectorNameFromVis] = useState(
+    formikToDetectorName(embeddable.vis.title)
+  );
+  const [intervalValue, setIntervalalue] = useState(10);
+  const [delayValue, setDelayValue] = useState(1);
+  const [enabled, setEnabled] = useState<boolean>(false);
+  const [shingleSizeValue, setShingleSizeValue] = useState(8);
```

Comment:
These values should be coming from formik fields instead of using react state. Then you can remove all of `on*Change()` fns below. You already are doing it for the features so you should be able to extend it to the rest of the values as well.

Basically the component can be wrapped in a `Field` with the formik field value and populated automatically. See https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ConfigureModel/components/AdvancedSettings/AdvancedSettings.tsx#L61 as an example

Reply:
Additionally all of these fields can be added as part of the validation, so any field errors etc is highlighted automatically when submitting. Example: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ConfigureModel/containers/ConfigureModel.tsx#L144

Reply:
a bit confused on this - if I remove the `useState` usage, then how do I set an initial value/state to the field? The example you mentioned will render an empty field instead of showing an initial value when load the page

Reply:
You can set defaults in `initialDetectorValue`. This is what you already pass anyways as the initial values in the `Formik` component (line 376).

Reply:
I still see shingle size state, lmk if this is part of a new commit.

Reply:
Got it - that makes sense.